File: output/GPT3.5-Turbo/xCodeEval_4000/C++/Go/980770e7a017f49149b45e925e6fa9d6.go
package main

import (
	"fmt"
	"math"
)

func main() {
	var n int
	fmt.Scan(&n)

	a := int64(-1)
	b := int64(-1)

	tttt := int(math.Sqrt(float64(n)))

	for i := 1; i <= tttt+9; i++ {
		if n%i == 0 {
			t := n / i

			zt := int(math.Sqrt(float64(t)))

			for j := 1; j <= zt+9; j++ {
				if t%j == 0 {
					tt := t / j

					if tt >= 1 {
						ttt := (tt + 2) * int64(i+1) * int64(j+2)

						if a == -1 {
							a = ttt
						} else if ttt < a {
							a = ttt
						}

						if b == -1 {
							b = ttt
						} else if ttt > b {
							b = ttt
						}
					}
				}
			}

			zz := i
			i = n / i

			t = n / i

			zt = int(math.Sqrt(float64(t)))

			for j := 1; j <= zt+9; j++ {
				if t%j == 0 {
					tt := t / j

					if tt >= 1 {
						ttt := (tt + 2) * int64(i+1) * int64(j+2)

						if a == -1 {
							a = ttt
						} else if ttt < a {
							a = ttt
						}

						if b == -1 {
							b = ttt
						} else if ttt > b {
							b = ttt
						}
					}
				}
			}

			i = zz
		}
	}

	fmt.Println(a - int64(n), b - int64(n))
}

File: output/GPT4o-Mini/xCodeEval_4000/C++/Java/980770e7a017f49149b45e925e6fa9d6.java
import java.util.*;

public class 980770e7a017f49149b45e925e6fa9d6_nc {
    static final int N = 100, M = 10;
    static long[][] F = new long[N][M], G = new long[N][M];
    static int n;

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        n = scanner.nextInt();

        long a = -1, b = -1;

        int tttt = (int) Math.sqrt(n);

        for (int i = 1; i <= tttt + 8; i++) {
            if (n % i == 0) {
                int t = n / i;
                int zt = (int) Math.sqrt(t);

                for (int j = 1; j <= zt + 8; j++) {
                    if (t % j == 0) {
                        long tt = t / j;

                        if (tt >= 1) {
                            long ttt = (tt + 2) * (i + 1) * (j + 2);

                            if (a == -1) a = ttt;
                            else a = Math.min(a, ttt);

                            if (b == -1) b = ttt;
                            else b = Math.max(b, ttt);
                        }
                    }
                }

                int zz = i;
                i = n / i;

                t = n / i;
                zt = (int) Math.sqrt(t);

                for (int j = 1; j <= zt + 8; j++) {
                    if (t % j == 0) {
                        long tt = t / j;

                        if (tt >= 1) {
                            long ttt = (tt + 2) * (i + 1) * (j + 2);

                            if (a == -1) a = ttt;
                            else a = Math.min(a, ttt);

                            if (b == -1) b = ttt;
                            else b = Math.max(b, ttt);
                        }
                    }
                }

                i = zz;
            }
        }

        System.out.println((a - n) + " " + (b - n));
    }
}
